mongodbatlas/resource_mongodbatlas_project_api_key.go

```diff
@@ -209,17 +209,20 @@ func resourceMongoDBAtlasProjectAPIKeyUpdate(ctx context.Context, d *schema.Reso
 			}
 		}
 
-		// Removing api_keys from the project
-		for _, apiKey := range removedAPIKeys {
+		// Removing projects assignments
+		for _, apiKey := range removedAssignments {
 			projectID := apiKey.(map[string]any)["project_id"].(string)
 			_, err := conn.ProjectAPIKeys.Unassign(ctx, projectID, apiKeyID)
+			if err != nil && strings.Contains(err.Error(), "GROUP_NOT_FOUND") {
+				continue // allows removing assignment for a project that has been deleted
```

Comment:
i would normally avoid continue in this situation, although your solution is probably more clear than:

if err != nil { 
    if !strings.Contains(err.Error(), "GROUP_NOT_FOUND") { // allows removing assignment for a project that has been deleted
				return diag.Errorf("error removing api_key(%s) from the project(%s): %s", apiKeyID, projectID, err)
}
}

Reply:
Yeah, went for this approach to make this case explicit and easy to read